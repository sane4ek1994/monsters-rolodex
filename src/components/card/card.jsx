import React from 'react'

import './card.styles.css'

export const Card = ({ monster }) => {
  const { id, name, email } = monster

  return (
    <div key={id} className='card-container'>
      <img src={`https://robohash.org/${id}?set=set2&size=180x180`} alt={`monsters ${name}`} />
      <h2>{name}</h2>
      <p className='card-email'>{email}</p>
    </div>
  )
}
