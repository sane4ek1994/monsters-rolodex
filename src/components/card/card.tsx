import { Monster } from '../../App'

import './card.styles.css'

type CardProps = {
  monster: Monster
}

export const Card = ({ monster }: CardProps) => {
  const { id, name, email } = monster

  return (
    <div key={id} className='card-container'>
      <img src={`https://robohash.org/${id}?set=set2&size=180x180`} alt={`monsters ${name}`} />
      <h2>{name}</h2>
      <p className='card-email'>{email}</p>
    </div>
  )
}
