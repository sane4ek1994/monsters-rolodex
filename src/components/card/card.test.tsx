import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Card } from './card'

it('expect to render Card component', () => {
  const cardComponent = shallow(
    <Card
      monster={{
        id: '',
        name: '',
        email: ''
      }}
    />
  )
  expect(toJson(cardComponent)).toMatchSnapshot()
})
