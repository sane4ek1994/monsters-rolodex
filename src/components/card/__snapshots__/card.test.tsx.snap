// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component 1`] = `
<div
  className="card-container"
  key=""
>
  <img
    alt="monsters "
    src="https://robohash.org/?set=set2&size=180x180"
  />
  <h2 />
  <p
    className="card-email"
  />
</div>
`;
